docs/contributing/Style-guide-for-Flutter-repo.md

```diff
@@ -831,21 +831,60 @@ really is, and tends to encourage "spaghetti" code (where distant components ref
 than "lasagna" code (where each section of the code is cleanly layered and separable).
 
 
-### Avoid using `extension`.
+### Avoid using `extension` methods.
 
 Extension methods are confusing to document and discover. To an end developer,
 they appear no different than the built-in API of the class, and discovering
 the documentation and implementation of an extension is more challenging than
-for class members.
+for class members. Prefer instead adding methods directly to relevant classes.
 
-Prefer instead adding methods directly to relevant classes. If that is not
-possible, create a method that clearly identifies what object(s) it works with
-and is part of.
+Only use `extension` methods when discoverability is not a concern:
 
-(A rare exception can be made for extensions that provide temporary workarounds
+- Extension methods might be implemented as temporary workarounds
 when deprecating features. In those cases, however, the extensions and all their
 members must be deprecated in the PR that adds them, and they must be removed
 in accordance with our deprecation policy.)
+- Additionally, private class fields
+[cannot be safely accessed](https://github.com/dart-lang/sdk/issues/51641)
+if the receiver is a subtype that implements the interface.\
+In these cases, it's better to house these fields in an `extension`
+or to restructure them into private top-level functions if possible.
```

Comment:
This seems like a really weird case. Maybe I'm not understanding it correctly? Do you have an example of this?

Accessing privates in other objects is generally dubious design in the first place...